### DISCUSSION TITLE
Anchor sharing between hololens apps without using Azure Spatial Anchors

### DISCUSSION THREAD

**Discussion by XavierWielemans** [microsoft/MixedRealityToolkit-Unity#11807] [Q&A]
Hello, I am trying to set up an off-line demo involving several hololens headsets worn by demo participants in the same physical location. The goal is for everyone to see the same virtual holograms at the same position and orientation in the physical space.

As it has to be offline, disconneted from the internet, Azure Spatial Anchors is not an option, unfortunately.

I have discovered "local anchor transfer" but it is no longer available since Unity 2019 (I am building my demo on top of a Unity 2022 customer project so it has to be Unity 2022).

I have come across XRAnchorTransferBatch (successor of fomer class used for local anchor transfer) but I am struggling to make it work.

Did anyone try that? Or does anyone have some knowledge on how I should proceed?

Any hint or tip would be most appreciated!

Cheers,

Xavier

**Comment by ThomasAdel1** [comment]
Hi @XavierWielemans,

I know your post is older, but I just stumbled across it and wanted to reply anyway. Maybe it will help others who are interested.

I have also recently needed an offline alternative to anchor sharing. What was sufficient for us was the manual transfer of anchors via the HoloLens device portal. 
Anchors, maps and the SpatialMappingFiles can be exported and imported under the System->MapManager tab. In our case, this exchange is sufficient. You can find everything you need under the following link:

[Using the Windows Device Portal #MapManager](https://learn.microsoft.com/en-us/windows/mixed-reality/develop/advanced-concepts/using-the-windows-device-portal#map-manager)

In our case, it was very important that all HoloLens devices were up to date and that the previous data was reset before importing the data. Otherwise, some anchors were placed incorrectly.

I don't know if there is an API for this procedure that can be accessed via a script. But maybe that gives you a clue.

Best regards

**Comment by XavierWielemans** [comment]
Hi Thomas, thanks a lot for your tips! In our case we finally gave up
trying to work around Azure Spatial Anchors. We found out that the www
connection was needed only for a short period of time during
cross-calibration and that, once headsets were correctly configured with
one another in the same physical environment, we could close the internet
access and go on with a purely local wifi link. Which was the right
compromise for us.
But thanks for sharing your solution! As you say, hopefully it will help
others coming across the same kind of problem in the future.
Cheers!
Xavier
